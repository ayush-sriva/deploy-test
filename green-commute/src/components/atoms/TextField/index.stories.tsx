import { ComponentStory, ComponentMeta } from '@storybook/react';
import CustomTextField from '.';

export default {
  title: 'Components/Atoms/TextField',
  component: CustomTextField,
} as ComponentMeta<typeof CustomTextField>;

const Template: ComponentStory<typeof CustomTextField> = (args) => <CustomTextField {...args} />;

export const DemoTextField = Template.bind({});

DemoTextField.args = {
  placeholderText: 'Enter your location',
  value: '',
  eventHandler: () => {},
};
