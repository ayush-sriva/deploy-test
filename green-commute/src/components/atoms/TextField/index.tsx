import { styled, TextField } from '@mui/material';
import responsiveTheme from '../../../theme';

interface InputProps {
  placeholderText: string;
  eventHandler: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  value: string;
}

const StyledTextField = styled(TextField)(() => ({
  width: '400px',
  height: '48px',
  [responsiveTheme.breakpoints.down('sm')]: {
    width: '200px',
    height: '24px',
  },
  borderRadius: '8px',
  '& label.Mui-focused': {
    color: responsiveTheme.palette.primary.semilight,
  },
  '& .MuiOutlinedInput-root': {
    '& fieldset': {
      borderStyle: 'solid',
      borderWidth: '1px',
      borderColor: responsiveTheme.palette.greyScale.dark,
    },

    '&:hover fieldset': {
      borderStyle: 'solid',
      borderWidth: '1px',
      borderColor: responsiveTheme.palette.info.dark,
      boxShadow: 'rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.23) 0px 3px 6px',
    },
    '&.Mui-focused fieldset': {
      borderStyle: 'solid',
      borderWidth: '1px',
      borderColor: responsiveTheme.palette.info.dark,
    },
  },
  input: {
    '&::placeholder': {
      fontFace: responsiveTheme.typography.caption,
      color: responsiveTheme.palette.secondary.light,
    },
  },
}));

const StyledInputField: React.FC<InputProps> = ({ placeholderText, eventHandler, value }) => {
  return <StyledTextField variant='outlined' value={value} placeholder={placeholderText} onChange={eventHandler}/>;
};

export default StyledInputField;
