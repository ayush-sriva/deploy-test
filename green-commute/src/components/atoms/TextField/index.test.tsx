import renderer from 'react-test-renderer';
import TextField from '.';

describe('Single Input Textfield', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TextField placeholderText={'Hello world'} eventHandler={()=>{}} value={''} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
