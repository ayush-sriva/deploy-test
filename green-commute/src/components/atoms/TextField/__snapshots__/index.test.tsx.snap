// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Input Textfield renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-omd3jb-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Hello world"
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
